src/main/java/org/gaul/s3proxy/CrossOriginResourceSharing.java

```diff
@@ -33,7 +33,7 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-final class CrossOriginResourceSharing {
+public class CrossOriginResourceSharing {
```

Comment:
Why remove `final`?  This causes Checkstyle failures.  Also don't you need the constructors to be `public`?